docs/madr/decisions/057-rework-mesh-routes-matching.md

```diff
@@ -0,0 +1,902 @@
+is obsolete as Route will select gateway, and we want to apply policy on a route. Is there a need to make it more specific than route? 
+For example when user creates route targeting MeshGateway:
+
+```yaml
+kind: MeshHTTPRoute
+metadata:
+  name: demo-app-route
+  namespace: kuma-system
+  labels:
+    kuma.io/origin: zone
+    kuma.io/mesh: default
+spec:
+  targetRef:
+    kind: MeshGateway
+    name: demo-app
+  to:
+    - targetRef:
+        kind: Mesh
+      rules:
+        - matches:
+            - path:
+                type: PathPrefix
+                value: "/"
+          default:
+            backendRefs:
+              - kind: MeshService
+                name: demo-app_kuma-demo_svc_5000
+```
+
+then when you create timeout that targets this route: 
+
+```yaml
+kind: MeshTimeout
+metadata:
+  name: demo-app-timeout
+  namespace: kuma-system
+  labels:
+    kuma.io/origin: zone
+    kuma.io/mesh: default
+spec:
+  to:
+    - targetRef:
+        kind: MeshHTTPRoute
+        name: demo-app-route
+      default:
+        requestTimeout: 5s
+```
+
+This route already will be created only on MeshGateway, so is there a need to specify anything else then a Mesh in `spec.targetRef`
+of timeout policy?
+
+#### Additional validation
+
+When referencing Mesh*Route we should validate policy configuration to make sure user is applying only configuration 
+that can be applied on route.
+
+### Affected policies
+
+At the moment we can only target `Mesh*Routes` in MeshTimeout policy. This MADR is based on MeshTimeout policy.
+Looking at what can be configured at Envoy route we can also apply this to our MeshRetry policy. Rate limit is also configurable on route,
+but we are configuring rate limit on inbound traffic, which does not apply to this MADR. `MeshLoadBalancingStrategy` also
+configures hashing per route and we should probably add support for routes in the future
+
+#### Producer/consumer model for targeting Mesh*Route in other policies
+
+For MeshRetry our producer/consumer makes sense since producer config is a suggestion from owner of the service. If consumer
+wants to override it because they don't want to retry, they just want to fail fast or the want to retry on bigger number of errors
+they can configure it by applying consumer policy.
+
+MeshRetry example:
+
+```yaml
+# Producer policy with retry on 500 HTTP error code
+apiVersion: kuma.io/v1alpha1
+kind: MeshRetry
+metadata:
+  name: producer-retry
+  namespace: backend-ns
+  labels:
+    kuma.io/mesh: default
+spec:
+  targetRef:
+    kind: MeshService
+    name: web
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+      default:
+        http:
+          numRetries: 3
+          backOff:
+            baseInterval: 10ms
+            maxInterval: 1s
+          retryOn:
+            - "500"
+---
+# Consumer policy with override to retry on all 5xx errors
```

Comment:
Is it a valid example? It looks to me like a producer policy,

Reply:
righ, the example was incorrect, I've updated it